modules/storages/app/workers/storages/manage_nextcloud_integration_events_job.rb

```diff
@@ -30,22 +30,23 @@ module Storages
   class ManageNextcloudIntegrationEventsJob < ApplicationJob
     include ManageNextcloudIntegrationJobMixin
 
-    DEBOUNCE_TIME = 5.seconds.freeze
+    THREAD_DEBOUNCE_TIME = 4.seconds.freeze
+    JOB_DEBOUNCE_TIME = 5.seconds.freeze
     THREAD_KEY = :manage_nextcloud_integration_events_job_debounce_happend_at
 
     queue_with_priority :above_normal
 
     def self.debounce
-      last_debounce_happend_at = Thread.current[THREAD_KEY]
-      if last_debounce_happend_at.blank? || Time.current > (last_debounce_happend_at + DEBOUNCE_TIME)
+      last_debounce_happend_at = RequestStore.store[THREAD_KEY]
+      if last_debounce_happend_at.blank? || Time.current > (last_debounce_happend_at + THREAD_DEBOUNCE_TIME)
         count = Delayed::Job
                   .where("handler LIKE ?", "%job_class: #{self}%")
```

Comment:
Alert! This will lead to minor catastrophe in the existing code before this PR already.

In SaaS the delayed jobs of all tenants are in a shared table in the public schema.
This code will delete other tenants' jobs as well.

To fix this we would need to be able to patch this line from the multitenancy plugin to add a `.where("handler LIKE ?", "%namespace: #{Apartment::Tenant.current}%")`.

But I dislike this solution overall. Can we not find something that could make due without having to mess with the delayed jobs table?

Reply:
So, it means we have it already for quite some time :(